### PR TITLE
Update/Improve Parser (including 7.3.5)

### PR DESCRIPTION
**Pull request by funjoker**
Updated some parsings for 7.3.5
Add Option to recalculate reputation discount on moneycost (not working with goblins)

also fixed crash when fusing pkt

Fusion (temp?) fix:
`Parallel.ForEach(groups, WriteGroup);`
to
`Parallel.ForEach(groups, new ParallelOptions { MaxDegreeOfParallelism = 1 }, WriteGroup);`

### CODE REVIEW

**Comment by Aokromes** [comment]
https://app.codacy.com/app/TrinityCore/WowPacketParser/pullRequest?prid=1500039

File: WowPacketParser/Saving/FusionBinaryPacketWriter.cs [hunk 1]
```diff
@@ -20,7 +20,7 @@ public static void Write(IEnumerable<Packet> packets, Encoding encoding)
             var groups = packets.GroupBy(p => p.Opcode);
 
             // since we have one file per opcode it's possible to parallelize groups writing
```
Comment (funjoker) [inline WowPacketParser/Saving/FusionBinaryPacketWriter.cs]: This comment is wrong. Fusion is not split.

File: WowPacketParser/Store/Objects/QuestTemplate.cs [hunk 2]
```diff
@@ -238,7 +241,7 @@ public sealed class QuestTemplate : IDataModel
         public RaceMask? AllowableRaces;
 
         [DBFieldName("AllowableRaces", TargetedDatabase.WarlordsOfDraenor)]
-        public int? AllowableRacesWod;
+        public long? AllowableRacesWod;
```
Comment (Shauren) [inline WowPacketParser/Store/Objects/QuestTemplate.cs]: its unsigned in db

File: WowPacketParser/Store/Objects/TrainerSpell.cs [hunk 3]
```diff
@@ -6,6 +6,15 @@ namespace WowPacketParser.Store.Objects
     [DBTableName("trainer_spell")]
     public sealed class TrainerSpell : IDataModel
     {
+        public uint[] ReqAbility;
+
+        public void ConvertToDBStruct()
```
Comment (Shauren) [inline WowPacketParser/Store/Objects/TrainerSpell.cs]: why is this needed? `DBFieldName` can be configured with `startAtZero`

File: WowPacketParserModule.V7_0_3_22248/Parsers/QuestHandler.cs [hunk 4]
```diff
@@ -691,5 +700,63 @@ public static void HandleQueryQuestRewardResponse(Packet packet)
...
+                        Floor = packet.ReadInt32("Floor", i, j),
+                        Priority = packet.ReadInt32("Priority", i, j),
+                        Flags = packet.ReadInt32("Flags", i, j),
+                        WorldEffectID = packet.ReadInt32("WorldEffectID", i, j),
+                        PlayerConditionID = packet.ReadInt32("PlayerConditionID", i, j)
+                    };
+
+                    questPoi.WoDUnk1 = packet.ReadInt32("WoDUnk1", i, j);
+
+                    int int13 = packet.ReadInt32("QuestPOIBlobPoint", i, j);
```
Comment (Shauren) [inline WowPacketParserModule.V7_0_3_22248/Parsers/QuestHandler.cs]: please dont copypaste these horrible anonymous names...

File: WowPacketParser/Store/Objects/TrainerSpell.cs (hunk 3 above)
Comment (funjoker) [reply WowPacketParser/Store/Objects/TrainerSpell.cs]: Otherwise it puts tons of UPDATES even if they are not necessary.

File: WowPacketParserModule.V7_0_3_22248/Parsers/QuestHandler.cs (hunk 4 above)
Comment (funjoker) [reply WowPacketParserModule.V7_0_3_22248/Parsers/QuestHandler.cs]: Any Suggestions?

Comment (Shauren) [reply WowPacketParserModule.V7_0_3_22248/Parsers/QuestHandler.cs]: ...you have the name for it on the right side of `=`
